api/app/src/test/kotlin/packit/integration/controllers/PacketControllerTest.kt

```diff
@@ -197,4 +204,45 @@ class PacketControllerTest : IntegrationTest()
             assertThat(it.get("name").textValue()).isEqualTo("artefact-types")
         }
     }
+
+    @Test
+    @WithAuthenticatedUser(authorities = ["packet.read"])
+    fun `streamZip streams a zip file`()
+    {
+        val paths = listOf(
+            "a_renamed_common_resource.csv",
+            "artefact1/artefact_data.csv",
+            "artefact1/excel_file.xlsx",
+            "artefact1/internal_presentation.pdf",
+            "artefact1/other_extensions.txt",
+            "data.csv",
+            "input_files/plot.png",
+            "orderly.R",
+            "presentation.html"
+        )
+        val encodedPaths = paths.joinToString(",") { URLEncoder.encode(it, StandardCharsets.UTF_8.toString()) }
+
+        val result: ResponseEntity<ByteArray> = restTemplate.exchange(
+            URI("/packets/$idOfDownloadTypesPacket3/zip?paths=$encodedPaths"),
+            HttpMethod.GET,
+            getTokenizedHttpEntity()
+        )
+        assertEquals(result.statusCode, HttpStatus.OK)
+        assertEquals(result.headers["Transfer-Encoding"]?.firstOrNull(), "chunked") // Header denoting streaming
+        assertEquals(result.headers.contentType.toString(), "application/zip")
+
+        // Read the stream into a zip file
+        val zipInputStream = ZipInputStream(result.body!!.inputStream())
+        val entries = mutableListOf<String>()
+        var entry: ZipEntry? = zipInputStream.nextEntry!!
+        while (entry != null) {
+            println("entry.name: " + entry.name)
+            entries.add(entry.name)
```

Comment:
Might be nice to check sizes here as well.

Reply:
Unfortunately, the getters 'size' and 'compressedSize' are both returning -1, which denotes an unknown size.

Reply:
Amazingly, this works:
```kt
val nextEntry = zipInputStream.nextEntry
println(entry.size)
println(entry.compressedSize)
entry = nextEntry
```

found here https://stackoverflow.com/questions/25602406/zipentry-unknown-size-although-set